data/data-pipeline/data_pipeline/tests/sources/historic_redlining/test_etl.py

```diff
@@ -0,0 +1,66 @@
+# pylint: disable=protected-access
+import pathlib
+import pandas as pd
+from data_pipeline.tests.sources.example.test_etl import TestETL
+from data_pipeline.etl.sources.historic_redlining.etl import (
+    HistoricRedliningETL,
+)
+
+
+class TestHistoricRedliningETL(TestETL):
+    _ETL_CLASS = HistoricRedliningETL
+
+    _SAMPLE_DATA_PATH = pathlib.Path(__file__).parents[0] / "data"
+    _SAMPLE_DATA_FILE_NAME = "HRS_2010.xlsx"
+    _SAMPLE_DATA_ZIP_FILE_NAME = "HRS_2010.zip"
+    _EXTRACT_TMP_FOLDER_NAME = "HistoricRedliningETL"
+
+    def setup_method(self, _method, filename=__file__):
+        """Invoke `setup_method` from Parent, but using the current file name.
+
+        This code can be copied identically between all child classes.
+        """
+        super().setup_method(_method=_method, filename=filename)
+
+    def test_extract_produces_valid_data(self, snapshot, mock_etl, mock_paths):
+        etl = self._setup_etl_instance_and_run_extract(
+            mock_etl=mock_etl,
+            mock_paths=mock_paths,
+        )
+        tmp_df = pd.read_excel(
```

Comment:
What's going on in this test? You load the input data as an excel and check the length -- but don't you mean to check the output data? 

Also, see question above about, can we just make a single test for shape that uses class attributes for the expected shape.

Reply:
Here's the test it overrides:
```    def test_extract_produces_valid_data(self, snapshot, mock_etl, mock_paths):
        """Tests the extract method.

        Here we are verifying that the data that we extract is "readable". I added a snapshot to be thorough,
        but @Jorge -- do you think this is necessary?
        """
        etl = self._setup_etl_instance_and_run_extract(
            mock_etl=mock_etl,
            mock_paths=mock_paths,
        )
        tmp_df = pd.read_csv(
            etl.get_tmp_path() / self._SAMPLE_DATA_FILE_NAME,
            dtype={etl.GEOID_TRACT_FIELD_NAME: str},
        )
        snapshot.snapshot_dir = self._DATA_DIRECTORY_FOR_TEST
        snapshot.assert_match(
            tmp_df.to_csv(index=False, float_format=self._FLOAT_FORMAT),
            self._EXTRACT_CSV_FILE_NAME,
        )

```

So it's really trying to verify that extract in this case correctly gets us a parsable excel file that transform can work on. I needed to override tho because it's excel and not CSV. I check the shape to verify the extracted file is roughly what we expect